Code/Tools/FBuild/FBuildCore/Helpers/SLNGenerator.cpp

```diff
@@ -189,12 +187,8 @@ void SLNGenerator::WriteProjectListings( const AString& solutionBasePath,
                 // get all the projects this project depends on
                 for ( const AString & dependency : deps.m_Dependencies )
                 {
-                    // For backward compatibility, keep the preceding slash and .vcxproj extension for GUID generation
-                    const char * projNameFromSlash = dependency.FindLast( NATIVE_SLASH );
-                    AStackString<> projectNameForGuid( projNameFromSlash ? projNameFromSlash : dependency.Get() );
-
                     AStackString<> newGUID;
-                    VSProjectGenerator::FormatDeterministicProjectGUID( newGUID, projectNameForGuid );
+                    VSProjectGenerator::FormatDeterministicProjectGUID( newGUID, dependency );
```

Comment:
This section seems wrong in that it always uses the path for the GUID and doesn't check if one has been explicitly set in the BFF file.